src/hooks/screen/useChatScreenHook.ts

```diff
@@ -197,10 +197,15 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
   };
 
   const handleUserName = (item): string => {
+    if (item?.user?.anon_user_info_emoji_code) {
+      return getOfficialAnonUsername(item?.user);
+    }
+
     if (item?.user?.username !== 'AnonymousUser') {
       return item?.user?.username;
     }
-    return `Anonymous ${anon_user_info_emoji_name}`;
+
+    return getOfficialAnonUsername(selectedChannel);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this function are significant as they introduce a new way of handling anonymous usernames. The use of `getOfficialAnonUsername` function improves the consistency of anonymous username generation across the application. However, it's important to ensure that the `item?.user` object passed to `getOfficialAnonUsername` has all the necessary properties that the function expects. If not, it could lead to unexpected behavior or errors.
```diff
-    if (item?.user?.username !== 'AnonymousUser') {
-      return item?.user?.username;
-    }
-   return `Anonymous ${anon_user_info_emoji_name}`;
+    if (item?.user?.anon_user_info_emoji_code) {
+      return getOfficialAnonUsername(item?.user);
+    }
+
+     if (item?.user?.username !== 'AnonymousUser') {
+       return item?.user?.username;
+     }
+
+    return getOfficialAnonUsername(selectedChannel);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->